src/parser/RegExpParser.mts

```diff
@@ -855,11 +1089,17 @@ export class RegExpParser {
       this.raise('Invalid code point');
     }
     let n = 0;
+    let oldN = 0;
     for (let i = 0; i < length; i += 1) {
       const c = this.source[this.position];
       if (isHexDigit(c)) {
         this.position += 1;
+        oldN = n;
         n = (n << 4) | Number.parseInt(c, 16);
+        if (oldN > n) {
+          // overflow
+          this.raise('Invalid hex digit');
+        }
```

Comment:
On this I'm going to be somewhat ignorant, but I wonder if you've taken into account the sign bit.  `1 << 31` < 0.

Reply:
I'm bad at bit calculation 😭, so I prefer the current code

Reply:
Well, it won't be a problem unless length can be 8 or more.  There's only one place where that might be possible: `maybeParseRegExpUnicodeEscapeSequence` at line 1128.

There, you're going to throw for any sequence greater than `0x10FFFF`, which would be 6 characters.  So you can probably just throw the same exception:

```typescript
      if (end - this.position > 6) {
        this.raise('Invalid code point');
      }
```

Reply:
Once you've done that, you can add an Assert call inside `scanHex` to make sure no one else breaks that rule in the future.

Reply:
Heck, I could probably write an evil test262 case for this.